test/utils/noderesourcetopologies/noderesourcetopologies.go

```diff
@@ -79,19 +92,29 @@ func GetUpdatedForNode(cli client.Client, ctx context.Context, ref nrtv1alpha2.N
 			klog.Errorf("cannot get the updated NRT object %s/%s", ref.Namespace, ref.Name)
 			return false, err
 		}
-		// very cheap test to rule out false negatives
-		if updatedNrt.ObjectMeta.ResourceVersion == ref.ObjectMeta.ResourceVersion {
-			klog.Warningf("NRT for %q resource version didn't change", ref.Name)
+		equalZones = isEqualNRT(ref, updatedNrt)
+		if equalZones {
 			return false, nil
 		}
-		equalZones = apiequality.Semantic.DeepEqual(ref.Zones, updatedNrt.Zones)
-		klog.Infof("NRT change: updated to %s", e2enrt.ToString(updatedNrt))
-		return !equalZones, nil
+		return true, nil
```

Comment:
I believe the condition here can be avoided and simply return with !isEqualNRT.., also because there are no logs specific to each case.

Reply:
right, this is a relic of past work. I'll simplify the code

Reply:
...and add logs. Note we need to bubble up and return `equalZones`, this is why we have the explicit assignment